packages/amazon-ssm-agent/amazon-ssm-agent.spec

```diff
@@ -17,10 +12,29 @@ Source0: %{gorepo}-%{version}.tar.gz
 Source1000: clarify.toml
 
 BuildRequires: %{_cross_os}glibc-devel
+Requires: %{name}(binaries)
 
 %description
 %{summary}.
 
+%package bin
+Summary: Remote management agent binaries
+Provides: %{name}(binaries)
+Requires: (%{_cross_os}image-feature(no-fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(fips) or %{name}-fips-bin)
+
+%description bin
+%{summary}.
+
+%package fips-bin
+Summary: Remote management agent binaries, FIPS edition
+Provides: %{name}(binaries)
+Requires: (%{_cross_os}image-feature(fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(no-fips) or %{name}-bin)
```

Comment:
What does this part prevent?

```
or %{name}-bin
```

Reply:
> What does this part prevent?
> 
> 
> 
> ```
> 
> or %{name}-bin
> 
> ```

It's meant to prevent the bin and fips-bin packages from both being installed, since only one set of binaries would end up in use.

It's a bit redundant but hypothetically nothing stops someone from adding `Provides: %{_cross_os}image-feature(fips)` to a random package in their variant that doesn't have that feature set. So even if that happened, this would force only one of the two packages to be installed.